### ISSUE TITLE
PMD bogus warnings for dependencies in the same project...

### ISSUE THREAD

**Issue by docbill** [ChuckJonas/vscode-apex-pmd#111]
**Describe the bug**
Up until I did an sfdx update this wos working.  Now vscode will display pmd warnings for anything not declared in the same
class file.  Methods, Objects, etc.   The message is that they do not exist.

**Apex PMD Output**
- Please make sure "Show StdErr" and "Show StdOut" Settings are enabled 
- I have no idea what that comment means.   It is just a tab in vscode, there is no settings uption...
```
[{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 66,
	"startColumn": 21,
	"endLineNumber": 66,
	"endColumn": 64
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 94,
	"startColumn": 72,
	"endLineNumber": 94,
	"endColumn": 79
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 99,
	"startColumn": 17,
	"endLineNumber": 99,
	"endColumn": 52
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry",
	"startLineNumber": 101,
	"startColumn": 56,
	"endLineNumber": 101,
	"endColumn": 61
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 119,
	"startColumn": 39,
	"endLineNumber": 119,
	"endColumn": 44
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry",
	"startLineNumber": 125,
	"startColumn": 42,
	"endLineNumber": 125,
	"endColumn": 47
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry",
	"startLineNumber": 127,
	"startColumn": 44,
	"endLineNumber": 127,
	"endColumn": 49
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry",
	"startLineNumber": 128,
	"startColumn": 33,
	"endLineNumber": 128,
	"endColumn": 38
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry",
	"startLineNumber": 130,
	"startColumn": 27,
	"endLineNumber": 130,
	"endColumn": 32
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 173,
	"startColumn": 12,
	"endLineNumber": 173,
	"endColumn": 85
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 176,
	"startColumn": 48,
	"endLineNumber": 176,
	"endColumn": 68
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 178,
	"startColumn": 9,
	"endLineNumber": 178,
	"endColumn": 67
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 180,
	"startColumn": 26,
	"endLineNumber": 180,
	"endColumn": 29
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entriesMapMap",
	"startLineNumber": 181,
	"startColumn": 13,
	"endLineNumber": 181,
	"endColumn": 26
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 182,
	"startColumn": 17,
	"endLineNumber": 182,
	"endColumn": 52
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 182,
	"startColumn": 55,
	"endLineNumber": 182,
	"endColumn": 92
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry",
	"startLineNumber": 183,
	"startColumn": 67,
	"endLineNumber": 183,
	"endColumn": 72
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry",
	"startLineNumber": 184,
	"startColumn": 20,
	"endLineNumber": 184,
	"endColumn": 25
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entry.Entity_Object__r",
	"startLineNumber": 185,
	"startColumn": 43,
	"endLineNumber": 185,
	"endColumn": 65
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 190,
	"startColumn": 21,
	"endLineNumber": 190,
	"endColumn": 65
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Invalid type: Trigger_Management_Entry__mdt",
	"startLineNumber": 192,
	"startColumn": 35,
	"endLineNumber": 192,
	"endColumn": 38
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entriesMap",
	"startLineNumber": 193,
	"startColumn": 25,
	"endLineNumber": 193,
	"endColumn": 35
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entries",
	"startLineNumber": 195,
	"startColumn": 21,
	"endLineNumber": 195,
	"endColumn": 28
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: entriesMap",
	"startLineNumber": 199,
	"startColumn": 16,
	"endLineNumber": 199,
	"endColumn": 26
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt",
	"startLineNumber": 204,
	"startColumn": 55,
	"endLineNumber": 204,
	"endColumn": 84
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt",
	"startLineNumber": 205,
	"startColumn": 54,
	"endLineNumber": 205,
	"endColumn": 83
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt",
	"startLineNumber": 206,
	"startColumn": 55,
	"endLineNumber": 206,
	"endColumn": 84
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt",
	"startLineNumber": 207,
	"startColumn": 54,
	"endLineNumber": 207,
	"endColumn": 83
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt",
	"startLineNumber": 208,
	"startColumn": 55,
	"endLineNumber": 208,
	"endColumn": 84
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt",
	"startLineNumber": 209,
	"startColumn": 54,
	"endLineNumber": 209,
	"endColumn": 83
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt",
	"startLineNumber": 210,
	"startColumn": 56,
	"endLineNumber": 210,
	"endColumn": 85
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.Entity_Object__c",
	"startLineNumber": 224,
	"startColumn": 54,
	"endLineNumber": 224,
	"endColumn": 100
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.Class_Name__c",
	"startLineNumber": 231,
	"startColumn": 23,
	"endLineNumber": 231,
	"endColumn": 66
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.Non_Entity_Object__c",
	"startLineNumber": 233,
	"startColumn": 23,
	"endLineNumber": 233,
	"endColumn": 73
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.Sequence_Number__c",
	"startLineNumber": 234,
	"startColumn": 23,
	"endLineNumber": 234,
	"endColumn": 71
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.SObjectType",
	"startLineNumber": 236,
	"startColumn": 32,
	"endLineNumber": 236,
	"endColumn": 73
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.Non_Entity_Object__c",
	"startLineNumber": 239,
	"startColumn": 31,
	"endLineNumber": 239,
	"endColumn": 81
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.Sequence_Number__c",
	"startLineNumber": 240,
	"startColumn": 36,
	"endLineNumber": 240,
	"endColumn": 84
},{
	"resource": "/home/remote/briemers/source/vscode/ApexFramework-fork/submodules/Apex-Framework/src/apexframework/main/default/classes/TriggerManagement.cls",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Variable does not exist: Trigger_Management_Entry__mdt.Non_Entity_Object__c",
	"startLineNumber": 286,
	"startColumn": 56,
	"endLineNumber": 286,
	"endColumn": 106
}]

```
**Apex PMD Settings**
- Pleas workspace & user) settings -->e provide the "EFFECTIVE" (check workspace & user) settings
-- what is the command line to provide this output?  Or what is the file name?

- pmdPath:
-- No clue

- rulesetPath:
-- No clue

**Apex File**
https://github.com/docbill/Apex-Framework/blob/main/src/apexframework/main/default/classes/TriggerManagement.cls

**Screenshots**
![Screenshot from 2022-08-11 08-45-33](https://user-images.githubusercontent.com/3169368/184136508-2ec4f8f8-08df-4050-8f86-532039909b11.png)

**Versions (please complete the following information):**
Version: 1.70.0
Commit: da76f93349a72022ca4670c1b84860304616aaa2
Date: 2022-08-04T04:38:48.541Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Linux x64 5.18.13-200.fc36.x86_64
